### ISSUE TITLE
Folder hierarchy is broken with gulp-typescript 2.13.0

### ISSUE THREAD

**Issue by carbonrider** [ivogabe/gulp-typescript#327]
The gulp-typescript should output .js files in respective folders (maintain hierarchy).

The folder hierarchy is lost and all the files are dumped in one location.

Here is the snippet of my **gulpfile.js**
```
var gulp = require('gulp');
var ts = require('gulp-typescript');
var sourcemaps = require('gulp-sourcemaps');
var path = require('path');

var tsProject = ts.createProject('tsconfig.json');

gulp.task('transpile', function(){
  return tsProject.src()
  .pipe(sourcemaps.init())
  .pipe(ts(tsProject))
  .js
  .pipe(sourcemaps.write('.', {
    includeContent: false,
    sourceRoot: function(file){
      return file.cwd;
    }
  }))
  .pipe(gulp.dest('./build'));
});

```
Here is the snippet of **tsconfig.json**
```
{
    "compilerOptions": {
        "module": "commonjs",
        "moduleResolution": "node",
        "target": "es5",
        "noImplicitAny": false,
        "noEmitHelpers": false,
        "experimentalDecorators": true,
        "sourceMap": true
    },
    "exclude": [
        "node_modules",
        "build",
        "config",
        "typings/browser",
        "typings/browser.d.ts"
    ]
}
```

The problem is occurring with gulp-typescript version 2.13.0. If I switch to older version 2.12.1, the folder hierarchy is maintained while generating js file.

**Comment by ivogabe** [comment]
Can you post your directory structure?

**Comment by carbonrider** [comment]
Thanks for your quick response.

Please check below image generated after using gulp-typescript 2.12.1
![image](https://cloud.githubusercontent.com/assets/5756215/14579619/6c8cf7e2-03cd-11e6-9a1b-8c16ec89084c.png)

And here is the image for gulp-typescript 2.13.0
![image](https://cloud.githubusercontent.com/assets/5756215/14579662/1865fc20-03cf-11e6-99b9-e76252fe1df8.png)

I just switched from older version to newer and started seeing this behavior.

**Comment by ivogabe** [comment]
It looks that the current behavior is correct. Older versions had a bug, which caused that the root directory was not calculated correctly. Older version considered the directory containing the `tsconfig.json` file as the project root, but `tsc` uses the common directory of all sources as the root. This bug has now been fixed. In your case, all sources are in `api/controllers/`, thus that is the root directory. `gulp.dest` works relative to this directory, so `api/controllers/search.js` is written as `build/search.js`. You can fix this by adding TypeScript files in other directories or by setting `"rootDir": "./"` in the compiler options.

**Comment by carbonrider** [comment]
Soon I will be having another directory adjacent to the controllers having similar names for **ts** files (modules). I guess, in that case the files will be overwritten with the gulp-typescript 2.13.0.
Also on the home page [https://github.com/ivogabe/gulp-typescript](https://github.com/ivogabe/gulp-typescript), the flag you have listed is said to be non-supported. Are you still suggesting the use of the flag.
In addition to above `api/controllers` is not the root directory for my code, in fact the file which contains tsconfig.json is the root. The folders specified in exclude section of `tsconfig.json` are actually adjacent to it.

**Comment by ivogabe** [comment]
The readme is outdated, thanks for pointing that out. It should say that it is only supported for usage with `tsProject.src()`. You can find more details on the bug in #253 and #276, it was fixed in 2.11.0 and another fix (related to projects that have definition files) in 2.12.2.
